shipping_addresses.go

```diff
@@ -0,0 +1,77 @@
+package recurly
+
+import (
+	"encoding/xml"
+)
+
+// ShippingAddress represents a shipping address
+type ShippingAddress struct {
+	XMLName       xml.Name `xml:"shipping_address"`
+	AccountCode   string   `xml:"account,omitempty"`
+	Subscriptions string   `xml:"subscriptions,omitempty"`
+	ID            string   `xml:"id,omitempty"`
+	FirstName     string   `xml:"first_name"`
+	LastName      string   `xml:"last_name"`
+	Nickname      string   `xml:"nickname,omitempty"`
+	Address       string   `xml:"address1"`
+	Address2      string   `xml:"address2"`
+	Company       string   `xml:"company,omitempty"`
+	City          string   `xml:"city"`
+	State         string   `xml:"state"`
+	Zip           string   `xml:"zip"`
+	Country       string   `xml:"country"`
+	Phone         string   `xml:"phone,omitempty"`
+	Email         string   `xml:"email,omitempty"`
+	VATNumber     string   `xml:"vat_number,omitempty"`
+	CreatedAt     NullTime `xml:"created_at,omitempty"`
+	UpdatedAt     NullTime `xml:"updated_at,omitempty"`
+}
+
+// UnmarshalXML unmarshals shipping addresses and handles intermediary state during unmarshaling
+// for types like href.
+func (s *ShippingAddress) UnmarshalXML(d *xml.Decoder, start xml.StartElement) error {
+	var v struct {
+		XMLName       xml.Name   `xml:"shipping_address"`
+		Account       hrefString `xml:"account,omitempty"`
+		Subscriptions hrefString `xml:"subscriptions,omitempty"`
+		ID            string     `xml:"id,omitempty"`
+		FirstName     string     `xml:"first_name"`
+		LastName      string     `xml:"last_name"`
+		Nickname      string     `xml:"nickname,omitempty"`
```

Comment:
The `Nickname` field isn't carried over below.

I think this repo could be simplified since this format you used is used elsewhere IIRC, but you can simplify the unmarshaling and avoid errors by creating an alias type and only creating the fields that  need to unmarshaled then casted to a separate type (the `hrefString` types). Will also be easier to maintain.

```go
// UnmarshalXML unmarshals shipping addresses and handles intermediary state during unmarshaling
// for types like href.
func (s *ShippingAddress) UnmarshalXML(d *xml.Decoder, start xml.StartElement) error {
	type shippingAddressAlias ShippingAddress
	var a struct {
		shippingAddressAlias
		XMLName       xml.Name   `xml:"shipping_address"`
		Account       hrefString `xml:"account,omitempty"`
	}
	if err := d.DecodeElement(&a, &start); err != nil {
		return err
	}
	*s = ShippingAddress(a.shippingAddressAlias)
	s.AccountCode = string(a.Account)
	return nil
}
```